DatadogCore/Tests/Datadog/DatadogCore/DatadogCoreTests.swift

```diff
@@ -243,4 +240,40 @@ class DatadogCoreTests: XCTestCase {
         XCTAssertEqual(storage2?.authorizedFilesOrchestrator.performance.maxFileAgeForWrite, 95)
         XCTAssertEqual(storage2?.authorizedFilesOrchestrator.performance.minFileAgeForRead, 105)
     }
+
+    func testWhenStoppingInstance_itDoesNotUploadEvents() throws {
+        // Given
+        let core = DatadogCore(
+            directory: temporaryCoreDirectory,
+            dateProvider: SystemDateProvider(),
+            initialConsent: .granted,
+            performance: .mockRandom(),
+            httpClient: HTTPClientMock(),
+            encryption: nil,
+            contextProvider: .mockAny(),
+            applicationVersion: .mockAny(),
+            maxBatchesPerUpload: .mockAny(),
+            backgroundTasksEnabled: .mockAny()
+        )
+
+        let requestBuilderSpy = FeatureRequestBuilderSpy()
+        try core.register(feature: FeatureMock(requestBuilder: requestBuilderSpy))
+        let scope = try XCTUnwrap(core.scope(for: FeatureMock.name))
+
+        // When
+        scope.eventWriteContext { context, writer in
+            writer.write(value: FeatureMock.Event(event: "should not be sent"))
+        }
+
+        core.stop()
```

Comment:
question/ This looks flaky ❄️ , doesn't it? If it takes more than "min file age" from  writing an event to stopping the core, data will be uploaded. What is the rationale here?

Reply:
Indeed, I can remove the first EWC. Rationale was to assert that no upload is triggered after stopping the core, event with event in the pipe.